src/Xamarin.Android.Tools.ApiXmlAdjuster/JavaApi.XmlModel.cs

```diff
@@ -24,15 +26,61 @@ public JavaPackage (JavaApi? parent)
 		{
 			Parent = parent;
 			
-			Types = new List<JavaType> ();
+			Types = new Dictionary<string, List<JavaType>> (StringComparer.OrdinalIgnoreCase);
 		}
 		
 		public  JavaApi?                Parent              { get; private set; }
 
 		public  string?                 Name                { get; set; }
 		public  string?                 JniName             { get; set; }
-		public  IList<JavaType>         Types               { get; set; }
-		
+
+		// Yes, there can be multiple types with the same *Java* name.
+		// For example:
+		// - MyInterface
+		// - MyInterfaceConsts
+		// It's debatable whether we handle this "properly", as most callers just
+		// do `First ()`, but it's been working for years so I'm not changing it.
+		// Exposes an IReadOnlyDictionary so caller cannot bypass our AddType/RemoveType code.
+		public  IReadOnlyDictionary<string, List<JavaType>>   Types               { get; }
+
+		// Use this for a flat list of *all* types
+		public IEnumerable<JavaType> AllTypes => Types.Values.SelectMany (v => v);
+
+		public void AddType (string name, JavaType type)
```

Comment:
As every instance of `AddType()` is `AddType(type.Name!, type)`, should this instead be `AddType(JavaType type)`, and below use `Types.TryGetValue(type.Name, out var list)`?

Reply:
Yeah, I was thinking there was a caller that set the `Name` property after adding the type to the collection, but that was a package, not a type.  Changed.